src/Coven.Agents/Streaming/AgentStreamSegmentationDaemon.cs

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: BUSL-1.1
+using System.Text;
+using Coven.Core;
+using Coven.Daemonology;
+
+namespace Coven.Agents.Streaming;
+
+public sealed class AgentStreamSegmentationDaemon(
```

Comment:
We should clean up the variable labeling/nullability here. I think this might need a manual rewrite just for cleanup.